solidity/core/test/validator-manager/outboxValidatorManager.test.ts

```diff
@@ -39,82 +84,314 @@ describe('OutboxValidatorManager', () => {
+    await helperOutbox.initialize(validatorManager.address);
   });
 
-  describe('#improperCheckpoint', () => {
+  describe('#prematureCheckpoint', () => {
+    const messageCount = 1;
+    // An premature checkpoint is one that has index greater than the latest index
+    // in the Outbox.
+    const prematureIndex = messageCount;
     const root = ethers.utils.formatBytes32String('test root');
-    const index = 1;
 
-    it('accepts an improper checkpoint if there is a quorum', async () => {
+    beforeEach(async () => {
+      for (let i = 0; i < messageCount; i++) {
+        await dispatchMessage(outbox, 'message');
+      }
+    });
+
+    it('accepts a premature checkpoint if it has been signed by a quorum of validators', async () => {
       const signatures = await signCheckpoint(
         root,
-        index,
-        [validator0, validator1], // 2/2 signers, making a quorum
+        prematureIndex,
+        [validator0, validator1], // 2/2 signers is a quorum
       );
 
       await expect(
-        validatorManager.improperCheckpoint(
+        validatorManager.prematureCheckpoint(
           outbox.address,
           root,
-          index,
+          prematureIndex,
           signatures,
         ),
       )
-        .to.emit(validatorManager, 'ImproperCheckpoint')
-        .withArgs(outbox.address, root, index, signatures);
+        .to.emit(validatorManager, 'PrematureCheckpoint')
+        .withArgs(
+          outbox.address,
+          root,
+          prematureIndex,
+          signatures,
+          messageCount,
+        );
       expect(await outbox.state()).to.equal(types.AbacusState.FAILED);
     });
 
-    it('reverts if there is not a quorum', async () => {
+    it('reverts if a premature checkpoint has not been signed a quorum of validators', async () => {
       const signatures = await signCheckpoint(
         root,
-        index,
+        prematureIndex,
         [validator0], // 1/2 signers is not a quorum
       );
 
       await expect(
-        validatorManager.improperCheckpoint(
+        validatorManager.prematureCheckpoint(
+          outbox.address,
+          root,
+          prematureIndex,
+          signatures,
+        ),
+      ).to.be.revertedWith('!quorum');
+    });
+
+    it('reverts if a non-premature checkpoint has been signed by a quorum of validators', async () => {
+      const validIndex = messageCount - 1;
+      const signatures = await signCheckpoint(
+        root,
+        validIndex,
+        [validator0, validator1], // 2/2 signers is a quorum
+      );
+
+      await expect(
+        validatorManager.prematureCheckpoint(
           outbox.address,
           root,
-          index,
+          validIndex,
+          signatures,
+        ),
+      ).to.be.revertedWith('!premature');
+    });
+  });
+
+  const dispatchMessagesAndReturnProofs = async (
+    differingIndex: number,
+    proofIndex: number,
+    messageCount: number,
```

Comment:
nit: take the parameters as an object so the callsite is more readable
```suggestion
  const dispatchMessagesAndReturnProofs = async ({
    differingIndex: number,
    proofIndex: number,
    messageCount: number,
    }
```